tools/generator/Java.Interop.Tools.Generator.ObjectModel/JavadocInfo.cs

```diff
@@ -210,6 +212,43 @@ static void AddNode (ICollection<string> comments, XNode node)
 			}
 		}
 
+		// Remove, replace, or decode common HTML tags to improve what is displayed in the IDE and online.
+		static string FixUpHtml (string javadocContent)
+		{
+			var replacements = new Dictionary<string, string> {
+				{ "&lt;blockquote&gt;", "" },    { "&lt;/blockquote&gt;", "" },
+				{ "&lt;cite&gt;", "<i>" },       { "&lt;/cite&gt;", "</i>" },
+				{ "&lt;code&gt;", "<c>" },       { "&lt;/code&gt;", "</c>" },
+				{ "&lt;dd&gt;", "" },            { "&lt;/dd&gt;", "" },
+				{ "&lt;dl&gt;", "" },            { "&lt;/dl&gt;", "" },
+				{ "&lt;dt&gt;", "" },            { "&lt;/dt&gt;", "" },
+				{ "&lt;em&gt;", "<i>" },         { "&lt;/em&gt;", "</i>" },
+				{ "&lt;h1&gt;", "" },            { "&lt;/h1&gt;", "" },
+				{ "&lt;h2&gt;", "" },            { "&lt;/h2&gt;", "" },
+				{ "&lt;h3&gt;", "" },            { "&lt;/h3&gt;", "" },
+				{ "&lt;h4&gt;", "" },            { "&lt;/h4&gt;", "" },
+				{ "&lt;h5&gt;", "" },            { "&lt;/h5&gt;", "" },
+				{ "&lt;h6&gt;", "" },            { "&lt;/h6&gt;", "" },
+				{ "&lt;li&gt;", "" },            { "&lt;/li&gt;", "" },
+				{ "&lt;ol&gt;", "" },            { "&lt;/ol&gt;", "" },
+				{ "&lt;strong&gt;", "<b>" },     { "&lt;/strong&gt;", "</b>" },
+				{ "&lt;sub&gt;", "" },           { "&lt;/sub&gt;", "" },
+				{ "&lt;sup&gt;", "" },           { "&lt;/sup&gt;", "" },
+				{ "&lt;table", "<table" },       { "&lt;/table&gt;", "</table>" },
+				{ "&lt;tbody&gt;", "<tbody>" },  { "&lt;/tbody&gt;", "</tbody>" },
+				{ "&lt;td&gt;", "<td>" },        { "&lt;/td&gt;", "</td>" },
+				{ "&lt;th&gt;", "<th>" },        { "&lt;/th&gt;", "</th>" },
+				{ "&lt;thead&gt;", "<thead>" },  { "&lt;/thead&gt;", "</thead>" },
+				{ "&lt;tr&gt;", "<tr>" },        { "&lt;/tr&gt;", "</tr>" },
+				{ "&lt;ul&gt;", "" },            { "&lt;/ul&gt;", "" },
+			};
+
+			foreach (var r in replacements) {
+				javadocContent = javadocContent.Replace (r.Key, r.Value);
```

Comment:
This is going to result in *lots* of garbage.  We should instead use a `StringBuilder` intermediary:

```csharp
var newContent = new StringBuilder (javadocContent);
foreach (var r in HtmlReplacements) {
    newContent.Replace (r.Key, r.Value);
}
return newContent.ToString ();
```

Reply:
I've removed this in favor of a regex term for parsing code elements as suggested.